app/controllers/products_controller.rb

```diff
@@ -0,0 +1,103 @@
+class ProductsController < ApplicationController
+  before_action :require_login, only: [:index, :edit, :new]
+  before_action :find_product, only: [:show, :edit, :retire, :activate]
+
+  def index
```

Comment:
When you have methods like this, you can put them on one line with `def index; end`!